#!/usr/bin/env python

PACKAGE = 'slam_karto'

from math import pi
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, int_t, double_t, str_t, bool_t

gen = ParameterGenerator()

gen.add("correlation_search_space_dimension", double_t, 0, "Correlation search space dimension.", 0.1, 0, 0.5)
gen.add("correlation_search_space_resolution", double_t, 0, "Correlation search space resolution.", 0.01, 0, 0.05)
gen.add("correlation_search_space_smear_deviation", double_t, 0, "Correlation search_space smear deviation.", .03, 0, .1)

gen.add("distance_variance_penalty", double_t, 0, "distance_variance_penalty.", .1, 0, 1.)
gen.add("angle_variance_penalty", double_t, 0, "angle_variance_penalty.", .1, 0, 1.)

gen.add("fine_search_angle_offset", double_t, 0, "fine_search_angle_offset.", 0.00349, 0, .1)
gen.add("coarse_search_angle_offset", double_t, 0, "coarse_search_angle_offset.", 0.349, 0, 1.)
gen.add("coarse_angle_resolution", double_t, 0, "coarse_search_angle_offset.", 0.0349, 0, .1)

gen.add("minimum_angle_penalty", double_t, 0, "minimum_angle_penalty.", .1, 0, 1.)
gen.add("minimum_distance_penalty", double_t, 0, "minimum_distance_penalty.", .1, 0, 1.)

gen.add("use_response_expansion", bool_t, 0, "Correlation search_space smear deviation.", True)

gen.add("rangeThreshold", double_t, 0, "rangeThreshold.", 1.0, 0, 100.)